Sources/NIO/System.swift

```diff
@@ -574,12 +562,11 @@ internal enum Posix {
     }
 }
 
-/// `NIOFailedToSetSocketNonBlockingError` indicates that NIO was unable to set a socket to non-blocking mode, either
-/// when connecting a socket as a client or when accepting a socket as a server.
+/// `NIOFailedToSetSocketCommand` indicates that NIO was unable to perform an
+/// operation on a socket.
 ///
-/// This error should never happen because a socket should always be able to be set to non-blocking mode. Unfortunately,
-/// we have seen this happen on Darwin.
-public struct NIOFailedToSetSocketNonBlockingError: Error {}
```

Comment:
We can't rename this error, it's a semver major. We should just add the new one instead. We can deprecate the `NIOFailedToSetSocketNonBlockingError` if we want to.

As for the new name, maybe `NIOFcntlFailedError`. Seems the most descriptive.

Reply:
Went the deprecation route.